pages/register.tsx

```diff
@@ -0,0 +1,251 @@
+import { Inter } from 'next/font/google';
+import { useRouter } from 'next/router';
+import { FormEvent, HTMLAttributes, useState } from 'react';
+import { twMerge } from 'tailwind-merge';
+import { Button } from '~/components/Button';
+import { Input } from '~/components/Input';
+import { Modal } from '~/components/Modal';
+import { TitleText } from '~/components/TitleText';
+import { SvgCheckMarkIcon } from '~/icons/CheckMarkIcon';
+import { createUser } from '~/utils/apiRequests';
+import { handleUserError } from '~/utils/handleError';
+import { emailRegex, passwordRegex } from '~/utils/regexPatterns';
+
+const inter = Inter({ subsets: ['latin'] });
+
+export default function Login() {
+  const [firstName, setFirstName] = useState('');
+  const [firstNameErrorText, setFirstNameErrorText] = useState('');
+  const [isFirstNameError, setIsFirstNameError] = useState(false);
+
+  const [lastName, setLastName] = useState('');
+  const [lastNameErrorText, setLastNameErrorText] = useState('');
+  const [isLastNameError, setIsLastNameError] = useState(false);
+
+  const [email, setEmail] = useState('');
+  const [emailErrorText, setEmailErrorText] = useState('');
+  const [isEmailError, setIsEmailError] = useState(false);
+
+  const [password, setPassword] = useState('');
+  const [passwordErrorText, setPasswordErrorText] = useState('');
+  const [isPasswordError, setIsPasswordError] = useState(false);
```

Comment:
Onko tämmöinen tapa ihan ok? Eli halusin jokaiselle input-kentälle oman virheen, jos annettu arvo on virheellinen. Mielestäni on parempi, kuin yksi "iso virhe". 

Voisiko tuota tehdä jotenkin geneerisesti, vai onko ok?

Reply:
Toinen vaihtoehto olisi tehdä yksi objekti jonka kenttiä nuo arvot ja virhee olisivat, mutta koska näitä on vain neljä niin näin helpompi ja selkeämpi.

Reply:
> Toinen vaihtoehto olisi tehdä yksi objekti jonka kenttiä nuo arvot ja virhee olisivat, mutta koska näitä on vain neljä niin näin helpompi ja selkeämpi.

Paljon parempi! Kiitos! :) :+1: